src/utils.js

```diff
@@ -226,6 +226,11 @@ function truncateString(str, maxStrLength = 10) {
   return `${str.slice(0, maxStrLength)}...`;
 }
 
+function urlValidation(url) {
```

Comment:
will be nice to have tests for various urls here esp. since it looks like we have a slightly custom url validation than what we may get with, say the URL() constructor (like requiring https which is awesome!) ? https://developer.mozilla.org/en-US/docs/Web/API/URL/URL